src/include/simeng/arch/aarch64/helpers/auxiliaryFunctions.hh

```diff
@@ -64,17 +64,39 @@ bitfieldManipulate(T value, T dest, uint8_t rotateBy, uint8_t sourceBits,
                    bool signExtend = false) {
   size_t bits = sizeof(T) * 8;
 
+  if (rotateBy >= bits) {
+    std::cerr << "\n[SimEng] Attempted to use a rotate amount of "
+              << unsigned(rotateBy)
+              << " in bitfieldManipulate which is greater than or equal to the "
+                 "data type size of "
+              << bits << "b in use. Exiting." << std::endl;
+    exit(1);
+  }
+
+  else if (sourceBits >= bits) {
+    std::cerr << "\n[SimEng] Attempted to use a source bit position "
+                 "value of "
+              << unsigned(sourceBits)
+              << " in bitfieldManipulate which is greater than or equal to the "
+                 "data type size of "
+              << bits << "b in use. Exiting." << std::endl;
+    exit(1);
+  }
```

Comment:
Given that the usage of this function comes exclusively from instruction execution, and the ISA has many in-built restrictions as to what values can be passed into the `rotateBy`, `sourceBits` and `bits` values, should these be `assert` rather than `if` statements?

Reply:
Thats a fair point. Lets leave it as is then. The print-outs should mention the file or function somewhere though (perhaps in the prefix) to make it more obvious where this error is coming from

Reply:
+1 to Finns last comment